### ISSUE TITLE
Still Giving Damaged error

### ISSUE THREAD

**Issue by Shoxjau** [firelad97/PROM-PCSX2-Running-on-Mac#1]
Heya!

I am still getting that damaged error even after disabling gatekeeper with: sudo spctl --master-disable

Any tips?

Thanks for all your hard work!

Shoxjau

**Comment by firelad97** [comment]
Thank you for posting this. I apologize, can you please tell me what OS version number you are on right now, again? It's for recording and testing purposes. Thanks! 

Once again, I will look into this issue myself and see what I can find. Although if I can't reproduce, we will have to wait for more people who have same issue, since I only have one computer here.

Thank you for your patience! ❤️

**Comment by Na0ture** [comment]
I also get this problem.It said that "PROM.app" was damaged error, 

- OS Version: macos Catalina 10.15.4

**Comment by Na0ture** [comment]
I solve this problem.

Step:
1. Open the terminal
2. Move to folder that had PROM.app by cd command
2. Run `sudo xattr -r -d com.apple.quarantine PROM.app`

**Comment by Shoxjau** [comment]
I am on 10.15.5

**Comment by tunghoang285** [comment]
> I solve this problem.
> 
> Step:
> 
> 1. Open the terminal
> 2. Move to folder that had PROM.app by cd command
> 3. Run `sudo xattr -r -d com.apple.quarantine PROM.app`

Na0ture can you explain step 2, thanks!

**Comment by Shoxjau** [comment]
So I just run that command? I'll give it a shot!

**Comment by iwbiz** [comment]
> I solve this problem.
> 
> Step:
> 
> 1. Open the terminal
> 2. Move to folder that had PROM.app by cd command
> 3. Run `sudo xattr -r -d com.apple.quarantine PROM.app`

This is perfect, thanks!

**Comment by firelad97** [comment]
@Na0ture Can you please where you get this from? I'd like to do research on this, thanks! And of course, thank you for sharing the solution!

**Comment by tunghoang285** [comment]
can anyone explain how to "Move to folder that had PROM.app by cd command"? I don't know anything about coding

**Comment by firelad97** [comment]
@tunghoang285 Type in `cd <directory>` where you replace `<directory>` with the folder where PROM is in. (or type just `cd` and then drag the folder where PROM is in to terminal)

**Comment by Na0ture** [comment]
@firelad97 [https://blog.obdev.at/gatekeeper-and-little-snitch-rules/](url)

also, I find something different from before and after when I use `sudo xattr -rd`. So I take a screenshot to easy to explain it.
![image](https://user-images.githubusercontent.com/26833533/83582085-1fb50300-a573-11ea-93fc-c2b48f178648.png)

**Comment by tunghoang285** [comment]
I did what Na0ture said, which got rid of the error message but now PROM won't even open at all

**Comment by firelad97** [comment]
@tunghoang285 that is an seperate issue, please make a new issue for that thank you

@na0ture, thank you! I'll be closing this issue now, assuming it fixes the OP's issue

**Comment by Shoxjau** [comment]
I am still having trouble. After running that command it tells me that no
program exists to run the .exe

**Comment by firelad97** [comment]
I will look into this issue this week. I will contact you in time.

**Comment by laggamer** [comment]
I ran into the same problem and I think that I know the solution.  The path should not be to the application that is PROM, but the place that PROM is located.  For example you want to in the path of downloads if PROM is in downloads.  I also had bash shell if that makes a difference or not.

**Comment by firelad97** [comment]
That is a little odd. I can't reproduce that @laggamer. Mind explaining more how exactly you made it happen, and how did bash shell helped you?

**Comment by laggamer** [comment]
I do not know if bash shell helped or not, but I am giving information that might be helpful.  I will show image of what I mean in order for you to better understand.

**Comment by laggamer** [comment]
I should of been more clear is that I followed the list of instructions by @Na0ture with the exception of using the downloads folder as the path.  The image below shows this and PROM opens.
![Screen Shot 2020-06-05 at 3 27 33 PM](https://user-images.githubusercontent.com/66453218/83920019-7a03be80-a741-11ea-9b61-28c2278cb088.jpeg)

**Comment by wh111msy** [comment]
same here ^^ where the original fix of disabling gatekeeper has no effect and the fix proposed itt has the same terminal response as lagggamers screenshot (pic below)

<img width="566" alt="Screen Shot 2020-06-05 at 6 53 37 PM" src="https://user-images.githubusercontent.com/56900030/83928456-decd1200-a75d-11ea-8054-28a7ebca304c.png">

**Comment by pmets44** [comment]
I followed @Na0ture's instructions, and I got PROM to open, but now it's just frozen when I open the First Time Configuration window and hit next.

EDIT: Solved! I command + clicked opened the folder, went to MacOS and opened the wineskin launcher instead of the app itself.

Second EDIT: PROM quit while I was trying to figure out how to get the BIOS working. I don't have a PS2 anymore and was just trying to get it to work with an online file. Now I am having the same problem as before, even when I open through the wineskin launcher

**Comment by Shoxjau** [comment]
Sorry, just got out of the hospital... I will try tommrow and follow up

**Comment by paintedsleeper** [comment]
<img width="574" alt="Screen Shot 2020-06-09 at 1 15 15 PM" src="https://user-images.githubusercontent.com/42075324/84184626-56978700-aa53-11ea-957b-b974c57e67ea.png">
having trouble with this process.

**Comment by iwbiz** [comment]
> <img alt="Screen Shot 2020-06-09 at 1 15 15 PM" width="574" src="https://user-images.githubusercontent.com/42075324/84184626-56978700-aa53-11ea-957b-b974c57e67ea.png">
> 
> having trouble with this process.

You can not cd into PROM.app, which is a file.
First check the directory you are in by entering `pwd`.
To check what is inside type `ls`.
To go back a directory do cd with 2 dots: `cd ..`
Now cd to your Downloads folder: `cd /Users/christasloan/Downloads/`
Then type `pwd` again and check if your're in the Downloads folder.
Continue the xattr operation...

**Comment by Gcenx** [comment]
@firelad97 I took a look over the majority of the issues, most `Wineskin` specific issues were resolved with the current Beta2 wrapper.

I have rebuilt PROM on a fork https://github.com/Gcenx/PROM-PCSX2-Running-on-Mac
As long as PROM is downloaded using Safari or the quarantine flag is removed it should launch and request the needed permissions when running macOS Catalina 10.15.4 or greater patch version without needing to disable SIP

For anyone running 10.15.0 > 10.15.3 SIP needs to be disabled, or update to 10.15.4 or greater patch version.

The `Visual C++ 2015-2019 x86 Redistributable` runtime packages were pre-installed using `winetricks` to avoid issues along with other verbs\
`WS11WineCX19.0.1-1` Engine was selected over `WS11WineCX64Bit19.0.1-1` as the emulator is 32Bit

In most cases PROM will request permission to access the needed directories, that or PROM can be added to `Full Disk Access`, that's more of a workaround in case of `tccd` database issues and it not recommended by Apple

**Comment by sjaykub** [comment]
@Gcenx, Didnt catalina drop 32bit?

**Comment by Gcenx** [comment]
@sjaykub yes macOS Catalina no longer ships with any 32Bit libraries and 32Bit code execution is disabled by default.

I’ve explained this numerous times but here we go again;
#### WS11WineCX19.0.1 contains
- wine (32Bit binary)
- wine32on64 (64Bit binary)
- wineserver (64Bit binary)
#### WS11WineCX64Bit19.0.1 contains
- wine (32Bit binary)
- wine32on64 (64Bit binary)
- wine64 (64Bit binary)
- wineserver (64Bit binary)

Now wine32on64 is a 64Bit binary that also links to 64Bit macOS libraries but runs 32Bit windows binaries, it needs a special entitlement for 10.15.0 > 10.15.3, from 10.15.4 the entitlement is no longer required (this also continues into macOS Big Sur beta on Intel systems)

**Comment by ruthcmoratagil** [comment]
Hi! Running Catalina 10.15.7 here, and unable to make it work. I run into the same issue mentioned above. Any clues on how to proceed?

![Captura de pantalla 2020-12-28 a las 13 19 27](https://user-images.githubusercontent.com/38922857/103213913-643d2d00-490f-11eb-9012-93ba1236bf2e.png)

**Comment by Gcenx** [comment]
@achtur this is using an older master wrapper release that had some issues, that could be upgraded to avoid at least the issue your seeing.

But it might be better to check the forums for the native 64Bit build for macOS as you might have better luck with that then using wine/32on64 to run the emulator.
